spec/language_field_spec.rb

```
@@ -0,0 +1,60 @@
+      solutions << RDF::Query::Solution.new(lang: RDF::URI('http://id.loc.gov/vocabulary/languages/spa'))
+      solutions << RDF::Query::Solution.new(lang: RDF::URI('http://id.loc.gov/vocabulary/languages/fre'))
+      lang_values = isd.send(:language_values)
+      expect(lang_values).to eq ['Italian', 'Spanish', 'French']
+    end
+    it 'uses raw language code if there is no translation in SEARCHWORKS_LANGUAGES' do
+      solutions << RDF::Query::Solution.new(lang: RDF::URI('http://id.loc.gov/vocabulary/languages/foo'))
+      solutions << RDF::Query::Solution.new(lang: RDF::URI('http://id.loc.gov/vocabulary/languages/1234'))
+      lang_values = isd.send(:language_values)
+      expect(lang_values).to eq ['foo', '1234']
+    end
+    it 'takes the language values from the Work, not the Instance' do
+      expect(sparql_conn).to receive(:query).with(a_string_matching('instanceOf')).and_return(solutions)
+      isd.send(:language_solutions)
+    end
+
+    context 'returns empty Array when' do
+      it 'there are no language values returned' do
+        lang_values = isd.send(:language_values)
+        expect(lang_values).to eq []
+      end
+      it 'language code is empty string' do
+        solutions << RDF::Query::Solution.new(lang: RDF::URI('http://id.loc.gov/vocabulary/languages/'))
+        lang_values = isd.send(:language_values)
+        expect(lang_values).to eq []
+      end
+      it 'sparql solution value is not URI' do
+        solutions << RDF::Query::Solution.new(lang: 'I am not a URI')
+        lang_values = isd.send(:language_values)
+        expect(lang_values).to eq []
+      end
+      it 'sparql solution is not URI beginning "http://id.loc.gov/vocabulary/languages/"' do
+        solutions << RDF::Query::Solution.new(lang: RDF::URI('http://something.else.org/ita'))
+        lang_values = isd.send(:language_values)
+        expect(lang_values).to eq []
+      end
```

Let's clarify the expected behavior for language URIs that are not LC URIs.  Can we expect to find such alternative URIs in the BF2 (or other) RDF data?  Should the code log an exception instead of silently passing it by?

---

I don't think we really need to worry about it at this point (if ever);  this test merely documents what the code is doing at this time.

---

@darrenleeweber will create an issue tracking this as a non optimal soultion

---

Documented in https://github.com/sul-dlss/sparql_to_sw_solr/issues/35